msal/src/Microsoft.Identity.Client/Platforms/iOS/UIParent.cs

```diff
@@ -0,0 +1,76 @@
+﻿//----------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+
+using Microsoft.Identity.Client.Internal;
+using Microsoft.Identity.Core.UI;
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Contains UI properties for interactive flows, such as the parent window (on Windows), or the parent activity (on Xamarin.Android), and 
+    /// which browser to use (on Xamarin.Android and Xamarin.iOS)
+    /// </summary> 
+    public sealed class UIParent
+    {
+        static UIParent()
+        {
+            ModuleInitializer.EnsureModuleInitialized();
+        }
+
+        internal CoreUIParent CoreUIParent { get; }
+
+        /// <summary>
+        /// Default constructor.
+        /// </summary>
+        public UIParent()
+        {
+            CoreUIParent = new CoreUIParent();
+        }
+
+        /// <summary>
+        /// Constructor for iOS for directing the application to use the embedded webview instead of the
+        /// system browser. See https://aka.ms/msal-net-uses-web-browser
+        /// </summary>
+        /// <remarks>This method is likely to be removed (replaced) before final release</remarks>
+        public UIParent(bool useEmbeddedWebview) : this()
+        {
+            CoreUIParent.UseEmbeddedWebview = useEmbeddedWebview;
+        }
+
+        /// <summary>
+        /// Platform agnostic constructor that allows building an UIParent from a NetStandard assembly.
```

Comment:
@henrik-me, @MarkZuber  - on iOS, this ctor is a bit out of place, because the "parent" param is always ignored. The same happens on net45 (useEmbeddedWebview is ignored), UWP (both params are ignored). However, the ctors need to exist to support calling the equivalent ctor from NetStanadrd. Otherwise, we'd get a MethodMissingException. 

Using the bait and switch technique (i.e. ref assemblies), we get the best of both worlds: 

- constructors exist at runtime, to allow creation from NetStandard 
- constructors are hidden at edit / build time if user programs directly against iOS so as not to increase public API / pollute the type

Reply:
Looks good.  CAn you add your comments from the PR here into the code itself?  I think that's a helpful reminder of what's going on here and it would be great to persist that knowledge into the codebase.